### PR TITLE
docs: add extended Fuzz Introspector doc

### PR DESCRIPTION
**Pull request by DavidKorczynski**
Fixes: https://github.com/google/oss-fuzz/issues/9444

Signed-off-by: David Korczynski <david@adalogics.com>

### CODE REVIEW

File: docs/advanced-topics/fuzz_introspector.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+parent: Advanced topics
+nav_order: 2
+permalink: /advanced-topics/fuzz-introspector/
+---
+
+# Fuzz Introspector
+{: .no_toc}
+
+For projects written in C/C++, Python and Java you can generate Fuzz
+Introspector reports to help guide the development of your fuzzing suit.
```
Comment (jonathanmetzman) [inline docs/advanced-topics/fuzz_introspector.md:13]: suite

File: docs/advanced-topics/fuzz_introspector.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+These reports help to extract details about the fuzzing set up of your
+project with the goal of making it easier to improve the fuzzing set up.
```
Comment (jonathanmetzman) [inline docs/advanced-topics/fuzz_introspector.md:15]: setup

File: docs/advanced-topics/fuzz_introspector.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+locally using the OSS-Fuzz helper script.
+
+
+- TOC
+{:toc}
+---
+
+## Fuzz Introspector overview
+
+As soon as your project is run with ClusterFuzz (< 1 day), you can view the Fuzz
```
Comment (jonathanmetzman) [inline docs/advanced-topics/fuzz_introspector.md:27]: <1

File: docs/advanced-topics/fuzz_introspector.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+a useful reference on how to make use of the reports.
+
+For ideas on how to use Fuzz Introspector, see [user guides](https://fuzz-introspector.readthedocs.io/en/latest/user-guides/index.html) which includes sections e.g.
+- [Quickly extract overview of a given project](https://fuzz-introspector.readthedocs.io/en/latest/user-guides/quick-overview.html)
+- [Get ideas for new fuzz targets](https://fuzz-introspector.readthedocs.io/en/latest/user-guides/get-ideas-for-new-targets.html)
+- [Comparing introspector reports](https://fuzz-introspector.readthedocs.io/en/latest/user-guides/comparing-introspector-reports.html)
+
+## Run Fuzz Introspector locally
+
+To generate a Fuzz Introspector report locally you can use `infra/helper.py`, which
```
Comment (jonathanmetzman) [inline docs/advanced-topics/fuzz_introspector.md:59]: complex wording, just say to use the `introspector` command.

File: docs/advanced-topics/fuzz_introspector.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+offers an `introspector` command. Fuzz Introspector relies on code coverage to
+analyse a given project, and this means we need to extract code coverage in the
```
Comment (jonathanmetzman) [inline docs/advanced-topics/fuzz_introspector.md:61]: Analyze. We prefer american spelling https://developers.google.com/style/spelling

File: docs/advanced-topics/fuzz_introspector.md [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+
+Assuming the above command is succesful you can view the report using `python3 -m http.server`
+following the example described above.
+
+
+## Differences in build tooling
+
+There are some differences in build environment for Fuzz Introspector builds
+in comparison to e.g. ASAN or code coverage builds. The reason is that
+Fuzz Introspector relies on various compile-time tooling to do its analysis.
```
Comment (jonathanmetzman) [inline docs/advanced-topics/fuzz_introspector.md:103]: various->certain.

**Review by jonathanmetzman — COMMENTED** [review]
LGTM with nits

Thanks for doing this.

**Review by jonathanmetzman — APPROVED** [review]
LGTM

**Review by oliverchang — APPROVED** [review]
nice!! thank you !!!

Can you please also edit https://github.com/google/oss-fuzz/blob/653827327a763ccc165537b4ff0d39b18a0fc63f/infra/build/build_status/update_build_status.py#L339 to add the link to the documentation to the index?

**Comment by DavidKorczynski** [comment]
Do we need trial runs for this?

**Pull request merged**